addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -339,12 +339,14 @@ def _compute_from_employee_id(self):
             elif not holiday.holiday_status_id and not holiday._origin.holiday_status_id:
                 holiday.holiday_status_id = default_holiday_status_id
 
-    @api.depends('holiday_status_id', 'allocation_type')
+    @api.depends('holiday_status_id', 'allocation_type', 'number_of_hours_display', 'number_of_days_display')
     def _compute_from_holiday_status_id(self):
         for allocation in self:
+            allocation.number_of_days = allocation.number_of_days_display
+            if allocation.number_of_hours_display:
+                allocation.number_of_days = allocation.number_of_hours_display / (allocation.employee_id.resource_calendar_id.hours_per_day or HOURS_PER_DAY)
+
```

Comment:
It's better to check if allocation.type_request_unit is == hour or days

Reply:
@jbm-odoo done